admin-manual.md

```diff
@@ -0,0 +1,187 @@
+[Flyway](https://flywaydb.org/) must also be present to migrate the databse as
+the application is developed, but it can be installed on a different server so
+long as it can access the database server.
+
+Defaults passwords should be changed.
+
+Once installed, start the mySQL console and create the database:
+
+    CREATE DATABASE lims;
+    USE lims;
+
+Then add a user that has all grant access on the 'lims' db:
+
+    GRANT ALL ON `lims`.* TO 'tgaclims'@'localhost';
+    GRANT ALL ON `lims`.* TO 'tgaclims'@'localhost' IDENTIFIED BY 'tgaclims';
+
+If your database and Tomcat install are on different machines, then you will
+obviously need to add a grant privilege to the MISO database from your remote
+machine:
+
+    GRANT ALL ON `lims`.* TO 'tgaclims'@'your.tomcat.install.server';
+    GRANT ALL ON `lims`.* TO 'tgaclims'@'your.tomcat.install.server' IDENTIFIED BY 'tgaclims';
+
+Download the Flyway command line tool and install it.
+
+# Setting Up the Applicaton Server
+The application server needs [Tomcat 8](https://tomcat.apache.org/download-80.cgi).
+
+Create a file called `ROOT.xml` in the following directory
+`$TOMCAT_ROOT/conf/Catalina/localhost`, creating the directory if necessary,
+and populate it with the following information:
+
+    <Context path="/ROOT" docBase="${catalina.home}/webapps/ROOT" debug="1">
+      <Resource name="jdbc/MISODB" type="javax.sql.DataSource"
+      driverClassName="com.mysql.jdbc.Driver"
+      initialSize="32"
+      maxIdle="10"
+      maxActive="100"
+      maxWait="1000"
+      removeAbandoned="true"
+      removeAbandonedTimeout="120"
+      logAbandoned="true"
+      testWhileIdle="true"
+      testOnBorrow="true"
+      testOnReturn="true"
+      validationQuery="select 1"
+      url="jdbc:mysql://localhost:3306/lims?autoReconnect=true&amp;zeroDateTimeBehavior=convertToNull&amp;useUnicode=true&amp;characterEncoding=UTF-8"
+      username="tgaclims"
+      password="tgaclims"/>
+    </Context>
+
+If your Tomcat install has the `autoDeploy="true"` flag set in `server.xml`, if
+you delete the `webapps/ROOT` directory and the `ROOT.war` file, Tomcat will
+delete the context `ROOT.xml` file. Either set autoDeploy to false, and
+manually deploy your webapp, or make the `ROOT.xml` file undeletable by using
+`chattr +i` (`chattr -i` will undo this operation). [Upstream
+bug](https://issues.apache.org/bugzilla/show_bug.cgi?id=40050)
+
+Set the database path:
+
+    url="jdbc:mysql://your.database.server:3306/lims"
+
+Download some supporting JARs:
+
+    cd $TOMCAT_ROOT/lib
+    curl -O https://repos.tgac.ac.uk/miso/common/mysql-connector-java-5.1.10.jar
+    curl -O https://repos.tgac.ac.uk/miso/common/jndi-file-factory-1.0.jar
+
+Append the following line to `$TOMCAT_ROOT/bin/setenv.sh` or, if using Tomcat from Debian or Ubuntu, `/etc/default/tomcat8`:
+
+    JAVA_OPTS="$JAVA_OPTS -Dsecurity.method=jdbc -Xmx768M"
+
+Create the directory `/storage/miso`.
```

Comment:
"Create the directory `/storage/miso` and download the default MISO configuration files."